### PR TITLE
feat: build app parameter settings [EXT-4837]

### PR DESCRIPTION
**Pull request by jjolton-contentful**
It is possible to specify [Parameter Definitions](https://www.contentful.com/developers/docs/extensibility/app-framework/app-parameters/#parameter-definition) on App Definitions. This PR provides a way to specify these installation/instance parameter schemas when running `create-app-definition`

### CODE REVIEW

File: package-lock.json
Comment (mgoudy91) [inline package-lock.json:1]: are the file changes on this intended?

File: packages/contentful--app-scripts/src/create-app-definition/build-app-parameter-settings.test.ts [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+    const parameters = await buildAppParameterSettings();
+
+    chai.expect(parameters.instance).to.deep.equal([
+      {
+        name: 'param1',
+        id: 'param1',
+        type: 'Symbol',
+        required: false,
+        default: undefined,
```
Comment (mgoudy91) [inline packages/contentful--app-scripts/src/create-app-definition/build-app-parameter-settings.test.ts:36]: are all these `x: undefined` necessary?

**Review by mgoudy91 — APPROVED** [review]
honestly looks great

File: packages/contentful--app-scripts/src/create-app-definition/build-app-parameter-settings.test.ts (hunk 1 above)
Comment (jjolton-contentful) [reply packages/contentful--app-scripts/src/create-app-definition/build-app-parameter-settings.test.ts:36]: yeah this is weird.. chai deep equal should work in theory without the undefineds but the tests fail if I omit them

File: package-lock.json
Comment (jjolton-contentful) [reply package-lock.json:1]: got this diff by running `npm run bootstrap`, thinking maybe I should revert this and cca package-lock.json

Comment (jjolton-contentful) [reply package-lock.json:1]: reverted 👍

**Review by msieroslawska — APPROVED** [review]

**Pull request merged**